lib/ansible/modules/cloud/misc/cloud_init_facts.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, René Moser <mail@renemoser.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cloud_init_facts
+short_description: Retrieve facts of cloud-init.
+description:
+  - Gathers facts by reading the status.json and result.json of cloud-init.
+version_added: 2.6
+author: René Moser (@resmo)
+options:
+  filter:
+    description:
+      - Filter facts
+    choices: [ status, result ]
+notes:
+  - See http://cloudinit.readthedocs.io/ for more information abount cloud-init.
+'''
+
+EXAMPLES = '''
+- name: Gather all facts of cloud init
+  cloud_init_facts:
+  register: result
+
+- debug:
+    var: result
+
+- name: Waiting for cloud init to finish
+  cloud_init_facts:
+    filter: status
+  register: res
+  until: not res.cloud_init_facts.status.v1.stage
+  retries: 50
+  delay: 5
+'''
+
+RETURN = '''
+---
+cloud_init_facts:
+  description: Facts of result and status.
+  returned: success
+  type: dict
+  sample: |
+    {
+      "status":{
+        "v1":{
+          "datasource":"DataSourceCloudStack",
+          "errors":[]
+        },
+        "result":{
+          "v1":{
+            "datasource":"DataSourceCloudStack",
+            "init":{
+              "errors":[],
+              "finished":1522066377.0185432,
+              "start":1522066375.2648022
+            },
+            "init-local":{
+              "errors":[],
+              "finished":1522066373.70919,
+              "start":1522066373.4726632
+            },
+            "modules-config":{
+              "errors":[],
+              "finished":1522066380.9097016,
+              "start":1522066379.0011985
+            },
+            "modules-final":{
+              "errors":[],
+              "finished":1522066383.56594,
+              "start":1522066382.3449218
+            },
+            "stage":null
+          }
+        }
+      }
+    }
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text
+
+
+CLOUD_INIT_PATH = "/var/lib/cloud/data/"
+
+
+class CloudInitFacts:
```

Comment:
This should be a new-style class (`CloudInitFacts(object)`) but doesn't really need to be a class since there's only one method. Simpler just to have a function.